pkg/code-generator/docgen/generator.go

```diff
@@ -32,26 +34,26 @@ var ignoredFiles = []string{
 }
 
 // write docs that are produced from the content of a single project
-func WritePerProjectsDocs(project *model.Project, genDocs *options.DocsOptions, absoluteRoot string) error {
-	if project.ProjectConfig.DocsDir != "" && (genDocs != nil) {
-		docs, err := GenerateFiles(project, genDocs)
-		if err != nil {
-			return err
+func WritePerProjectsDocs(project *model.Project, docsOptions *options.DocsOptions, absoluteRoot string) error {
+	if project.ProjectConfig.DocsDir != "" && docsOptions != nil {
+
+		if docsOptions.Output == "" {
+			docsOptions.Output = options.Markdown
 		}
 
-		for _, file := range docs {
-			path := filepath.Join(absoluteRoot, project.ProjectConfig.DocsDir, file.Filename)
-			if err := os.MkdirAll(filepath.Dir(path), os.ModePerm); err != nil {
-				return err
-			}
-			if err := ioutil.WriteFile(path, []byte(file.Content), 0644); err != nil {
-				return err
-			}
+		docGenerator := &DocsGen{
+			DocsOptions: *docsOptions,
+			Project:     project,
+			DocsDir:     filepath.Join(absoluteRoot, project.ProjectConfig.DocsDir),
 		}
+
+		return GenerateAndWriteFiles(docGenerator)
 	}
 	return nil
 }
 
+// DEPRECATED
+// prefer GenerateAndWriteFiles
```

Comment:
Is this still being used somewhere?

Reply:
It's not, but because it's a public function, I didn't want to delete it.